### PR TITLE
generic.tests: add case boot vm memory in hugepage

### PR DESCRIPTION
**Pull request by xutian**
ID: 1395544

Signed-off-by: Xu Tian <xutian@redhat.com>

### CODE REVIEW

**Comment by suqinhuang** [comment]
we can set it in the acceptance loop file

**Comment by xutian** [comment]
you can pickup this case in your acceptance loop

**Comment by CongLi** [comment]
How about hugepage_size for different archs?

**Comment by xutian** [comment]
No need to config it,  it will get by get_hugepage_size() if "setup_hugepages=yes", Please check test_setup.py.

**Comment by suqinhuang** [comment]
1. it will effect lots of loops, as reboot case is added in all the loops
2. put in acceptance loop will be more benefit if we want to "add, drop, update" cases for huge test

**Comment by xutian** [comment]
sounds good, will try

**Comment by qunfang** [comment]
@xutian after applied this patch and re-run reboot test, don't find the hugepage related parameter:

MALLOC_PERTURB_=1  /usr/libexec/qemu-kvm \
    -S  \
    -name 'avocado-vt-vm1'  \
    -sandbox off  \
    -machine pseries  \
    -nodefaults  \
    -vga std  \
    -chardev socket,id=qmp_id_qmpmonitor1,path=/var/tmp/avocado_ccvk0S/monitor-qmpmonitor1-20161117-022541-JtGNE0h8,server,nowait \
    -mon chardev=qmp_id_qmpmonitor1,mode=control  \
    -chardev socket,id=qmp_id_catch_monitor,path=/var/tmp/avocado_ccvk0S/monitor-catch_monitor-20161117-022541-JtGNE0h8,server,nowait \
    -mon chardev=qmp_id_catch_monitor,mode=control  \
    -chardev socket,id=serial_id_serial0,path=/var/tmp/avocado_ccvk0S/serial-serial0-20161117-022541-JtGNE0h8,server,nowait \
    -device spapr-vty,reg=0x30000000,chardev=serial_id_serial0 \
    -device pci-ohci,id=usb1,bus=pci.0,addr=03 \
    -drive id=drive_image1,if=none,snapshot=off,aio=native,cache=none,format=qcow2,file=/root/test_home/qzhang/staf-kvm-devel/workspace/usr/share/avocado/data/avocado-vt/images/RHEL-Server-7.3-ppc64le-virtio.qcow2 \
    -device virtio-blk-pci,id=image1,drive=drive_image1,bootindex=0,bus=pci.0,addr=04 \
    -device virtio-net-pci,mac=9a:d6:d7:d8:d9:da,id=idx6dOWK,vectors=4,netdev=idbsVO2Q,bus=pci.0,addr=05  \
    -netdev tap,id=idbsVO2Q,vhost=on,vhostfd=21,fd=20 \
    -m 8192  \
    -smp 4,maxcpus=4,cores=2,threads=1,sockets=2 \
    -device usb-tablet,id=usb-tablet1,bus=usb1.0,port=1  \
    -vnc :2  \
    -rtc base=utc,clock=host  \
    -boot order=cdn,once=c,menu=off,strict=off \
    -enable-kvm

**Comment by xutian** [comment]
@qunfang  please run case boot_vm_in_hugepage

**Comment by xutian** [comment]
@suqinhuang  split it to a new  case, it will not impact anybody.

**Comment by qunfang** [comment]
LGTM, ack'ed

**Comment by xutian** [comment]
@CongLi  any comments ?

File: generic/tests/cfg/boot_vm_in_hugepage.cfg [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    kill_vm_on_error = yes
+    login_timeout = 240
+    reboot_count = 25
+    slots_mem = 4
+    maxmem_mem = 32G
+    setup_hugepages = yes
+    mem_devs = hugepage
+    size_mem_hugepage = 256M
+    backend_mem = memory-backend-file
+    mem-path = /mnt/kvm_hugepage
```
Comment (CongLi) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: should be mem_path? is it a typo?

**Comment by CongLi** [comment]
Acked-by: Cong Li <coli@redhat.com>

**Pull request merged**